### PR TITLE
Add some information on tag headers

### PR DESCRIPTION
**Pull request by Mimickal**
Noticed we don't have any information on the header tags share

### CODE REVIEW

File: src/data/structs/h1/tags/header.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+entry_type: header
```
Comment (Mimickal) [inline src/data/structs/h1/tags/header.yml:1]: This file isn't currently used anywhere. I couldn't quite figure out how to get this to render the same way other tag parts do. It looks like that might take some actual JS work to get showing up properly.

Either way, I included it here for posterity.

File: src/content/h1/tags/readme.md [hunk 2]
```diff
@@ -46,4 +48,20 @@ Invalid tags can often be corrected by resetting fields and re-saving the tag us
...
+
+All tags have a common header. This header makes up the first 64 bytes of data, and contains the following fields:
+
+| Field       | Type     | Comments |
+|-------------|----------|----------|
+| padding     | pad      | 36 bytes |
+| tag group   | `uint32` | 4 character tag group ID encoded as an int |
+| checksum    | `uint32` |          |
+| header size | `uint32` | Default: 64 |
+| flags       | `uint64` | A 64 bit bitfield |
```
Comment (num0005) [inline src/content/h1/tags/readme.md:61]: I want to say only MEK uses that as a bitfield and not the official tools but I'm not certain of that.

File: src/content/h1/tags/readme.md [hunk 3]
```diff
@@ -28,6 +28,8 @@ Tag paths also appear in arguments to [Tool][] and scripting.
 ## Blocks
 A _tag block_ field, also known as a _reflexive_, is essentially a list of smaller data structures within a tag. An example is the [scenario][] tag containing a block of vehicle spawns points. In visual tag editors, blocks appear as a list of elements which are often editable by adding or removing elements. A block field internally consists of an item count and a pointer to an array of structures of the expected type.
 
+The term _reflexive_ is considered outdated and comes from early [halo-map-tools][] modding.
```
Comment (num0005) [inline src/content/h1/tags/readme.md:31]: 👍

File: src/content/h1/tags/readme.md [hunk 4]
```diff
@@ -46,4 +48,20 @@ Invalid tags can often be corrected by resetting fields and re-saving the tag us
...
+All tags have a common header. This header makes up the first 64 bytes of data, and contains the following fields:
+
+| Field       | Type     | Comments |
+|-------------|----------|----------|
+| padding     | pad      | 36 bytes |
+| tag group   | `uint32` | 4 character tag group ID encoded as an int |
+| checksum    | `uint32` |          |
+| header size | `uint32` | Default: 64 |
+| flags       | `uint64` | A 64 bit bitfield |
+| version     | `uint16` | Default: 1 |
```
Comment (num0005) [inline src/content/h1/tags/readme.md:62]: Not sure how I feel about this being described in the same way that `header size` is, that almost universally is 64 (I think the tag reading code in the official tools is even hardcoded to assume that), while `version` iirc is set to whatever the definition version was set to - it just happens to be incremented from 1.

**Review by csauve — APPROVED** [review]
I rebased this and pushed through some conflicts. All the info is added and I incorporated some header info from invader's code too.

**Comment by Mimickal** [comment]
Oh thank you! I forgot I had this PR open

**Pull request closed without merging**